plugins/rest/rest_auth.go

```diff
@@ -457,6 +458,11 @@ func (ap *awsSigningAuthPlugin) NewClient(c Config) (*http.Client, error) {
 
 func (ap *awsSigningAuthPlugin) Prepare(req *http.Request) error {
 	logrus.Debug("Signing request with AWS credentials.")
-	err := signV4(req, ap.awsCredentialService(), time.Now())
+	// Assume we're calling s3 unless specified otherwise
+	service := "s3"
+	if ap.AWSService != nil {
```

Comment:
This should probably be moved to the `NewClient` construct as there's no need to check for this on each request. 
```go
if ap.AWSService == nil {
    ap.AWSService = "s3"
}
```